### ISSUE TITLE
"there was a problem parsing the package" message

### ISSUE THREAD

**Issue by darkonije** [StefanIlchev/ElementumService#23]
Hello Stefan.
Thank you for a good work.
I see that for many of us Elementum works on android 10+, but I still have an issue with it. Whatever I do this message is on my screen when trying to run apk. And it doesn't matter if it is universal or specific one. Too many times too many tries.
Simply I don't know what to do.
I have some android 10 tv box.
Sorry to bother you.

**Comment by StefanIlchev** [comment]
This was reported [before](https://github.com/StefanIlchev/ElementumService/issues/16) and all I could think of was a corrupt download. Since then however I've encountered something that might be similar. To help test this, could you please try installing the attached modified builds of another one of my [repositories](https://github.com/StefanIlchev/CallBlocker) and report which if any you were able to. I chose to make test builds out of that repository instead of this one because its APKs are smaller.

[CallBlocker.zip](https://github.com/StefanIlchev/ElementumService/files/10285014/CallBlocker.zip)

**Comment by darkonije** [comment]
Yes, I've seen that message before and because of your answer I've tried to
download everything on multiple devices, put it on usb.
attached apks are on my usb and will try to install those later.
I'm going to try kodi and elementum now on my phones.
an update is comming.

чет, 22. дец 2022. у 09:30 StefanIlchev ***@***.***> је
написао/ла:

**Comment by darkonije** [comment]
elementum apk installed on two phones (android 11) easily.
will try callblocker apks when i get home

чет, 22. дец 2022. у 09:43 Darko Nije ***@***.***> је написао/ла:

**Comment by darkonije** [comment]
Nope. Not even one of CallBlocker releases doesn't work. Same message from
the system "there was a problem parsing the package"

**Comment by StefanIlchev** [comment]
Can you please confirm that there are apps that you can install from APK files.

Maybe logcat can tell us what exactly is the problem. Try the following:

1. clear logcat:

```bat
adb logcat -c
```

2. install the app:

```bat
adb install "CallBlocker-release-10-type-C.apk"
```

3. dump log:

```bat
adb logcat -d
```

4. share the output of these commands.

**Comment by StefanIlchev** [comment]
If you can find an open source app like Kodi itself or VLC that you can install from a file, I'll check what they're doing differently.

**Comment by darkonije** [comment]
Kodi is installed from apk, and some other apps like media player for
example.

**Comment by darkonije** [comment]
Not even one. So sorry for bothering you.

**Comment by StefanIlchev** [comment]
No problem, I'm curious what your device doesn't like about the way I build my projects :-)

I have other ideas but guessing blindly is taking too long. If you or someone else would be willing to connect to such a device with Android Debug Bridge (ADB) we might get a more meaningful error message.

**Comment by darkonije** [comment]
Yes, my device is a real disaster.
I am willing to do that thing with ADB but I ha e to find out how to do
that. I am going to do some research and hope for the best. 😁

**Comment by StefanIlchev** [comment]
Maybe it wants the APK to be signed as a JAR file, like an Android 6 or older device.

[ElementumService2.zip](https://github.com/StefanIlchev/ElementumService/files/10298863/ElementumService2.zip)

**Comment by yasarozg** [comment]
Same problem here, tried all the zips including ElementumService2.zip got the same error message.
